From fdbbdc4ff0dee43aa4a878d11daea2870499a1b5 Mon Sep 17 00:00:00 2001
From: Lev Stipakov <lev@openvpn.net>
Date: Mon, 29 Apr 2019 10:26:13 +0300
Subject: [PATCH 3/6] error_code.ipp: Use English for Windows error messages

FormatMessageA doesn't work well with non-ASCII chars
so make it return error message in English.

Signed-off-by: Lev Stipakov <lev@openvpn.net>
Signed-off-by: Frank Lichtenheld <frank@lichtenheld.com>
---
 asio/include/asio/impl/error_code.ipp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/asio/include/asio/impl/error_code.ipp b/asio/include/asio/impl/error_code.ipp
index caeb1967c..4f5f6ed8e 100644
--- a/asio/include/asio/impl/error_code.ipp
+++ b/asio/include/asio/impl/error_code.ipp
@@ -80,7 +80,7 @@ public:
     DWORD length = ::FormatMessageA(FORMAT_MESSAGE_ALLOCATE_BUFFER
         | FORMAT_MESSAGE_FROM_SYSTEM
         | FORMAT_MESSAGE_IGNORE_INSERTS, 0, value,
-        MAKELANGID(LANG_NEUTRAL, SUBLANG_DEFAULT), (char*)&msg, 0, 0);
+        MAKELANGID(LANG_ENGLISH, SUBLANG_ENGLISH_US), (char*)&msg, 0, 0);
     detail::local_free_on_block_exit local_free_obj(msg);
     if (length && msg[length - 1] == '\n')
       msg[--length] = '\0';
-- 
2.34.1

